ocaml/tests/test_ha_vm_failover.ml

```diff
@@ -569,4 +681,813 @@ module ComputeMaxFailures = Generic.MakeStateful (struct
       ]
 end)
 
-let tests = [("plan_for_n_failures", PlanForNFailures.tests)]
+let extract_output_for_anti_affinity_plan __context pool anti_affinity_plan =
+  let slv1 =
+    Db.Host.get_all ~__context
+    |> List.find (fun self -> Db.Host.get_name_label ~__context ~self = slave1)
+  in
+  let slave1_vms_increasing =
+    Db.Host.get_resident_VMs ~__context ~self:slv1
+    |> List.map (fun self -> (self, Db.VM.get_record ~__context ~self))
+    |> List.filter (fun (_, record) -> not record.API.vM_is_control_domain)
+    |> List.map (fun (self, record) ->
+           (self, Xapi_ha_vm_failover.vm_memory ~__context record)
+       )
+    |> List.sort Xapi_ha_vm_failover.bigger_than
+  in
+  let hosts_increasing =
+    Db.Host.get_all ~__context
+    |> List.filter (fun h ->
+           Db.Host.get_name_label ~__context ~self:h <> "localhost"
+       )
+    |> List.filter (( <> ) slv1)
+    |> List.map (fun host ->
+           ( host
+           , Memory_check.host_compute_free_memory_with_maximum_compression
+               ~__context ~host None
+           )
+       )
+    |> List.sort Xapi_ha_vm_failover.bigger_than
+  in
+  let vm_can_boot_on_host vm host =
+    let slv1 = pool.slaves |> List.find (fun s -> s.name_label = slave1) in
+    let slave1_vm =
+      slv1.vms
+      |> List.find (fun v ->
+             v.vm_name_label = Db.VM.get_name_label ~__context ~self:vm
+         )
+    in
+    not
+      (List.exists
+         (fun h -> h = Db.Host.get_name_label ~__context ~self:host)
```

Comment:
We'd better assign `Db.Host.get_name_label ~__context ~self:host` to a variable. Then we could call the ``Db.Host.get_name_label ` only once.

Reply:
`Db.Host.get_name_label` is called 4 times in this function(`extract_output_for_anti_affinity_plan`), but they are called for different host.